src/ui/public/index_patterns/index_patterns.js

```diff
@@ -41,6 +41,7 @@ export function IndexPatternsProvider(Notifier, Private) {
   self.cache = patternCache;
   self.getIds = getProvider('id');
   self.getTitles = getProvider('attributes.title');
+  self.getIndexPatterns = getProvider();
```

Comment:
I agree that the naming here isn't ideal and it's making this PR harder than it has to be.

These three methods return something different than the other `.get()` in here. The latter appears to be fetching index patterns from saved objects, but also [potentially updating the data from ES](https://github.com/elastic/kibana/blob/master/src/ui/public/index_patterns/_index_pattern.js#L221). The former is just fetching from saved objects cache. I'm not even sure if this distinction matters at this point, but there is a comment in [_get.js](https://github.com/elastic/kibana/blob/master/src/ui/public/index_patterns/_get.js#L8): 
> // didn't incorporate with the indexPattern cache to prevent id collisions.

I'm guessing the separate cache exists for a good reason so we should probably leave that as is for now. However, understanding the difference might make it easier to figure out the naming here. I'm still not sure what's the best name for all these things.

Paging @spalger for some insight, since it seems [he added that comment](https://github.com/elastic/kibana/commit/efe34132d369ca1b1d5e955d2b4effa2f8149ca8#diff-022f6b8718f8174673c52e05447bbde8)